File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        hs = set()

        longest = 0

        l, r = 0, 0
        while r < len(s):
            if s[r] in hs:
                hs.remove(s[l])
                l += 1
            else:
                longest = max(longest, r - l + 1)
                hs.add(s[r])
                r += 1
        return longest
                


        

File: Data Structures & Algorithms/longest-common-subsequence/submission-0.java
class Solution {
    public int longestCommonSubsequence(String text1, String text2) {
        int rows = text1.length();
        int cols = text2.length();
        int dp[][] = new int[rows + 1][cols + 1];
        for (int i = rows - 1; i >= 0; i--) {
            for (int j = cols - 1; j >= 0; j--) {
                if (text1.charAt(i) == text2.charAt(j)) {
                    dp[i][j] = dp[i + 1][j + 1] + 1;
                } else {
                    dp[i][j] = Math.max(dp[i + 1][j], dp[i][j + 1]);
                }
            } 
        }
        return dp[0][0];
    }
}
